src/Game.java

```diff
@@ -0,0 +1,316 @@
+    public void changeMoney() {
+        moneyLabel.setText("Coins: " + character.getMoney());
+        moneyLabel.setCenter(moneyButton.getCenter());
+    }
+
+    /**This method makes sure that the character is on top of all the objects on screen. */
+    public void makeCharacterOnTop() {
+        character.removeFromCanvas(canvas);
+        character.addToCanvas(canvas);
+    }
+
+    /**This method allows a player to harvest plants by pressing the space bar*/
+    public void harvestByKey(KeyboardEvent event) {
+        Plant plant = character.getPlantAtPosition(character.getPosition());
+        if (plant == null) {
+            return;
+        }
+        if (event.getKey() == Key.SPACE && plant.maxSize()) {
+            character.harvest(canvas, plant);
+            landplot.minusOne();
+            makeCharacterOnTop();
+            changeMoney();
+        }
+    }
+
+    /**
+     * This method allows the user to plant crops by pressing different keys. A crop can only be planted
+     * if it is unlocked. if they have not reached the maximum number of plants for a landplot, and if
+     * they can afford it.
+     * @param event
+    */
+    public void plant(KeyboardEvent event) {
+        double x = character.getX();
+        double y = character.getY();
+        landplot = environment.getLandPlotAtPosition(character.getPosition());
+    
+
+        if (landplot == null) {
+            return;
+        }
+
+        if (event.getKey() == Key.W 
+            && character.getMoney() >= 10
+            && !landplot.isLocked() 
+            && !appleLock&&landplot.getCnt()<landplot.getMaximumPlants()) {  
+                         
+            character.plantApple(canvas, x, y);
+            makeCharacterOnTop();
+            changeMoney();
+            landplot.plusOne();
+        }
+        if (event.getKey() == Key.E
+            && character.getMoney() >= 15 
+            && !landplot.isLocked() 
+            && !orangeLock&&landplot.getCnt()<landplot.getMaximumPlants()) {
+                
+            character.plantOrange(canvas, x, y);
+            makeCharacterOnTop();
+            changeMoney();
+            landplot.plusOne();
+        }
+        if (event.getKey() == Key.Q
+            && character.getMoney() >= 5
+            && !landplot.isLocked()
+            &&landplot.getCnt()<landplot.getMaximumPlants()) {
+                
+            character.plantPotato(canvas, x, y);
+            makeCharacterOnTop();
+            changeMoney();
+            landplot.plusOne();
+        }
+        if (event.getKey() == Key.R 
+            && character.getMoney() >= 20 
+            && !landplot.isLocked() 
+            && !cabbageLock&&landplot.getCnt()<landplot.getMaximumPlants()) {
+                
+            character.plantCabbage(canvas, x, y);
+            makeCharacterOnTop();
+            changeMoney();
+            landplot.plusOne();
+        }
```

Comment:
Again, lots of repetition here! Map from `Key` → `Plant`, then have a single piece of code handle all of these cases.